config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/InboundJmsDestinationDetail.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.beans;
+
+import static java.util.Arrays.stream;
+
+public class InboundJmsDestinationDetail extends JmsDestinationDetail {
+    
+    private AcknowledgeType acknowledgeType;
+    private ServiceResolutionSettings serviceResolutionSettings;
+    private String failureQueueName;
+    private boolean isEnabled;
+    private Integer numOfConsumerConnections;
+    private Integer maxMessageSizeBytes; // null = default, -1 = Do not override,  0 = Unlimited.
+    
+    public InboundJmsDestinationDetail() {
+        super();
+    }
+
+    public AcknowledgeType getAcknowledgeType() {
+        return acknowledgeType;
+    }
+
+    public void setAcknowledgeType(AcknowledgeType acknowledgeType) {
+        this.acknowledgeType = acknowledgeType;
+    }
+
+    public ServiceResolutionSettings getServiceResolutionSettings() {
+        return serviceResolutionSettings;
+    }
+
+    public void setServiceResolutionSettings(ServiceResolutionSettings serviceResolutionSettings) {
+        this.serviceResolutionSettings = serviceResolutionSettings;
+    }
+
+    public String getFailureQueueName() {
+        return failureQueueName;
+    }
+
+    public void setFailureQueueName(String failureQueueName) {
+        this.failureQueueName = failureQueueName;
+    }
+
+    public boolean isEnabled() {
+        return isEnabled;
+    }
+
+    public void setIsEnabled(boolean enabled) {
+        isEnabled = enabled;
+    }
+
+    public Integer getNumOfConsumerConnections() {
+        return numOfConsumerConnections;
+    }
+
+    public void setNumOfConsumerConnections(Integer numOfConsumerConnections) {
+        this.numOfConsumerConnections = numOfConsumerConnections;
+    }
+
+    public Integer getMaxMessageSizeBytes() {
+        return maxMessageSizeBytes;
+    }
+
+    public void setMaxMessageSizeBytes(Integer maxMessageSizeBytes) {
+        this.maxMessageSizeBytes = maxMessageSizeBytes;
+    }
+
+    public static class ServiceResolutionSettings {
+        private String serviceRef;
+        private String soapActionMessagePropertyName;
+        private ContentTypeSource contentTypeSource;
+        private String contentType;
+        
+        public ServiceResolutionSettings() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. ([squid:S1186](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1186))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWekObWnbde-piqdGWhL&open=AWekObWnbde-piqdGWhL)

[ISSUE_KEY]: AWekObWnbde-piqdGWhL